AK/Bitmap.h

```diff
@@ -163,69 +157,151 @@ class Bitmap {
         return {};
     }
 
-    Optional<size_t> find_longest_range_of_unset_bits(size_t max_length, size_t& found_range_size) const
+    /**
+     * the function is supposed to find the next range of unset bits 
+     * starting from the @from value. 
+     * @from: the postition from which the search starts. The var
+     *        will be changed and new value is the offset of the
+     *        found block
+     * @min_length: minimum size of the range which will be returned
+     * @max_length: shows when to stop adding unset bits to the range.
+     *            This is used for optimisation, since the range of
+     *            unset bits can be long, and we don't need the whole
+     *            size, so we can stop when we've reached @max_length.
+     * It's recomended to use @min_length and @max_length, since they
+     * give a boost.
+     */
+    inline Optional<size_t> find_next_range_of_unset_bits(size_t& from, size_t min_length=1, size_t max_length=0xffffffff) const
     {
-        auto first_index = find_first_unset();
-        if (!first_index.has_value())
-            return {};
 
-        size_t free_region_start = first_index.value();
-        size_t free_region_size = 1;
+#ifdef AK_ARCH_I386
+        #define trailing_zeroes_cnt(x) (__builtin_ctz(x))
+#else
+        auto trailing_zeroes_cnt = [](u32 val) -> u32{
+            for (u8 i = 0; i < 32; ++i) {
+                if ((val >> i) & 1) {
+                    return i;
+                }
+            }
+            return 0;
+        };
+#endif /* AK_ARCH_I386 */
 
-        size_t max_region_start = free_region_start;
-        size_t max_region_size = free_region_size;
+        if (min_length > max_length) {
+            return {};
+        }
 
-        // Let's try and find the best fit possible
-        for (size_t j = first_index.value() + 1; j < m_size && free_region_size < max_length; j++) {
-            if (!get(j)) {
-                if (free_region_size == 0)
-                    free_region_start = j;
-                free_region_size++;
-            } else {
-                if (max_region_size < free_region_size) {
-                    max_region_size = free_region_size;
-                    max_region_start = free_region_start;
+        u32 *bitmap32 = (u32*)m_data;
+        
+        // calculating the start offset
```

Comment:
[Please write comments in proper English.](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#comments)